webpack.config.js

```diff
@@ -14,18 +25,407 @@ for (var i = 0; i < baseViewPaths.length; i++) {
+//   - Include base project .po files
+// =============================================================================
+
+
+// (1) Set up paths to files and directories needed for the build, as well as
+//     constants and utilities
+// -----------------------------------------------------------------------------
+
+const VERBOSE = true; // Controls logging output
+const PORT = 8000;
+
+// Flavor base
+var flavorBasePath = path.resolve(
+  __dirname, 
+  "src/flavors",
+  process.env.FLAVOR
+);
+var flavorConfigPath = path.resolve(
+  flavorBasePath,
+  "config.yml"
+);
+
+// Handlebars templates
+var baseJSTemplatesPath = path.resolve(
+  __dirname,
+  "src/base/jstemplates"
+);
+var flavorJSTemplatesPath = path.resolve(
+  flavorBasePath,
+  "jstemplates"
+);
+var flavorPagesPath = path.resolve(  // NOTE: pages are a flavor-only concept,
+                                     // so there's no basePagesPath
+  flavorBasePath,
+  "jstemplates/pages"
+);
+var outputJSTemplatesPath = path.resolve(
+  __dirname,
+  "src/base/static/dist/jstemplates"
+);
+var compiledTemplatesOutputPath = path.resolve(
+  __dirname,
+  "src/base/jstemplates/templates.js" // TODO: is this the right place for this?
+);
+
+// Handlebars executable
+var handlebarsExec = path.resolve(
+  __dirname,
+  "node_modules/handlebars/bin/handlebars"
+);
+
+// Images and markers
+var baseImageAssetsPath = path.resolve(
+  __dirname,
+  "src/base/static/css/images"
+);
+var flavorImageAssetsPath = path.resolve(
+  flavorBasePath,
+  "static/css/images"
+);
+var outputImageAssetsPath = path.resolve(
+  __dirname,
+  "src/base/static/dist/images"
+);
+
+// Localization
+var baseLocaleDir = path.resolve(
+  __dirname,
+  "src/base/locale"
+);
+var flavorLocaleDir = path.resolve(
+  flavorBasePath,
+  "locale"
+);
+
+const PO_FILE_NAME = "django.po";     // Assumes all flavors will have a .po 
+                                      // file matching this name
+const CONFIG_GETTEXT_REGEX = /^_\(/;
+const JSTEMPLATES_GETTEXT_REGEX = /{{#_}}(.*?){{\/_}}/g;
+const BASE_STATIC_URL = "/static/";
+const MAPBOX_TOKEN = process.env.MAPBOX_TOKEN;
+
+// TODO: add this to .env
+const CLICKY_ANALYTICS_ID = process.env.CLICKY_ANALYTICS_ID || "";
+
+// TODO: add this to .env
+const MAPQUEST_KEY = process.env.MAPQUEST_KEY || "";
+
+// TODO: add this to .env
+const GOOGLE_ANALYTICS_ID = process.env.GOOGLE_ANALYTICS_ID || "";
+
+// Pull out dataset urls from the .env file. We ignore the keys, as they're no
+// longer needed.
+var datasetSiteUrls = {};
+Object.keys(process.env).forEach(function(key) {
+  if (key.endsWith("_SITE_URL")) {
+    datasetSiteUrls[key] = process.env[key];
+  }
+});
+
+// Logging
+var log = function(msg, time) {
+  if (VERBOSE) {
+    console.log(
+      "(STATIC SITE BUILD) ", 
+      msg, 
+      (
+        (time)
+          ? "in " + time + " ms" 
+          : ""
+      )
+    );
+  }
+};
+
+// Timing
+var time = function(fn) {
```

Comment:
We should probably get rid of this timing function - I don't see the benefit of having it in the master branch. It's fine for taking benchmarks, but otherwise it's creating extra complexity through the use of callbacks.

Also, you can try making this an `async` function, and calling it with the `await` keyword. That should save you from using the callbacks.